### DISCUSSION TITLE
PieChart: Support number fields as name for value

### DISCUSSION THREAD

**Discussion by calebedwards21** [grafana/grafana#41180] [Dashboards, Panels & Visualizations]
When trying to create a pie chart in Grafana 8.2.2, I cannot use a number as a metric name. Reference the below picture.
The title to this is wrong. Should say 'Metric Names as Numbers'.

![incorrect-pie-chart](https://user-images.githubusercontent.com/31447494/139751226-d503fb99-089f-4b43-99d0-7c08fa5a1357.PNG)

When using strings as metric names the pie chart works just fine. I just need to be able to use the pie chart when I have metric names that are numbers. The title to this picture should be 'Metric Names as Strings'.
![correct-pie-chart](https://user-images.githubusercontent.com/31447494/139751232-c4d2d13e-b38f-41e6-bc26-5f50ebc82966.PNG)

**Comment by torkelo** [comment]
In order for Grafana to know what field to use as "name" it will only look for string fields in this case. So this data format is currently not supported

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
